aws-s3-transfer-manager/src/operation/download/body.rs

```diff
@@ -2,13 +2,89 @@
  * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
  * SPDX-License-Identifier: Apache-2.0
  */
-use crate::operation::download::service::ChunkResponse;
-use aws_smithy_types::byte_stream::AggregatedBytes;
+use aws_smithy_types::byte_stream::ByteStream;
+use bytes::{Buf, Bytes};
+use bytes_utils::SegmentedBuf;
 use std::cmp;
 use std::cmp::Ordering;
 use std::collections::BinaryHeap;
+use std::io::IoSlice;
 use tokio::sync::mpsc;
 
+use crate::error::ErrorKind;
+
+use super::chunk_meta::ChunkMetadata;
+
+/// Non-contiguous Binary Data Storage
+///
+/// When data is read from the network, it is read in a sequence of chunks that are not in
+/// contiguous memory. [`AggregatedBytes`](crate::byte_stream::AggregatedBytes) provides a view of
+/// this data via [`impl Buf`](bytes::Buf) or it can be copied into contiguous storage with
+/// [`.into_bytes()`](crate::byte_stream::AggregatedBytes::into_bytes).
+#[derive(Debug, Clone)]
+pub struct AggregatedBytes(SegmentedBuf<Bytes>);
```

Comment:
Also this is maybe fine for now but is this how we want to expose the data/chunks? I originally did it via `AggregatedBytes` since that was basically the only option to do what we wanted with the API given from the SDK without introducing a new API (which we are now doing and should). 

I don't have a suggestion at the moment but it bears consideration since we are effectively taking control over the API and that may give us different options than originally considered in the initial "naive" implementation.

Reply:
Thanks, moved to io. `AggregatedBytes` seems fine to me for exposing the body. As discussed offline, we can batch up multiple ChunkResponse as a future optimization.